### PR TITLE
[bitnami/fluentd] Allow to configure persistence for forwarders

### PR DESCRIPTION
**Pull request by forselli-stratio**
**Description of the change**

Allow to configure persistence for forwarders

**Benefits**

In some scenarios, if you don't configure persistence for forwarders you may face some data loss.

**Possible drawbacks**

NONE

### CODE REVIEW

File: bitnami/fluentd/README.md [hunk 1]
```diff
@@ -84,6 +84,7 @@ The following tables lists the configurable parameters of the fluentd chart and
 | `forwarder.service.loadBalancerSourceRanges`    | Addresses that are allowed when service is LoadBalancer                                                        | `[]`                                                                                                    |
 | `forwarder.service.clusterIP`                   | Static clusterIP or None for headless services                                                                 | `nil`                                                                                                   |
 | `forwarder.service.annotations`                 | Annotations for the forwarder service                                                                          | `{}`                                                                                                    |
+| `forwarder.persistence.enabled`                 | Enable persistence volume for the forwarder                                                                          | `false`                                                                                                    |
```
Comment (juan131) [inline bitnami/fluentd/README.md:87]: I'm missing `forwarder.persistence.hostPath.path` parameter here

File: bitnami/fluentd/values.yaml [hunk 2]
```diff
@@ -291,6 +291,20 @@ forwarder:
...
+    enabled: false
+    ## If persistence enabled, a hostPath volume mounts a directory of your choice from the host node's filesystem into your Pod.
+    ## The host directory you chose is mounted into /opt/bitnami/fluentd/logs/buffers in your Pod
+    ## Example use case: mount host directory /tmp/buffer (if the directory doesn't exist, it creates it) into forwarder pod.
+    ##   persistence:
+    ##     enabled: true
+    ##     hostPath: 
+    ##       path: /tmp/buffer
+    hostPath: 
+      path: /opt/bitnami/fluentd/logs/buffers
```
Comment (juan131) [inline bitnami/fluentd/values.yaml:306]: Doesn't this stutter `hostPath.path`?

File: bitnami/fluentd/templates/forwarder-daemonset.yaml [hunk 3]
```diff
@@ -111,8 +111,13 @@ spec:
           volumeMounts:
             - name: fluentd-config
               mountPath: /opt/bitnami/fluentd/conf
+            {{- if .Values.forwarder.persistence.enabled }}
+            - name: {{ include "fluentd.fullname" . }}-buffer
```
Comment (juan131) [inline bitnami/fluentd/templates/forwarder-daemonset.yaml:115]: I'd use:
```suggestion
            - name: fluentd-buffer
```
I mean, I don't see any advantage on having this name based on the `fluentd.fullname` macro.

**Review by juan131 — CHANGES_REQUESTED** [review]
Thanks so much for the PR! Please ensure you rebase your PR from master since there are some conflicts.
Please also take a look to my suggestions.

File: bitnami/fluentd/README.md (hunk 1 above)
Comment (forselli-stratio) [reply bitnami/fluentd/README.md:87]: You're right, added. Thanks.

File: bitnami/fluentd/values.yaml (hunk 2 above)
Comment (forselli-stratio) [reply bitnami/fluentd/values.yaml:306]: Agree, but i wanted to keep the same structure kubernetes uses in hostPath volumes: https://kubernetes.io/docs/concepts/storage/volumes/#hostpath 
I can change change it necessary though. Thanks

Comment (juan131) [reply bitnami/fluentd/values.yaml:306]: Fair enough

**Review by juan131 — APPROVED** [review]
LGTM

**Pull request merged**